scripts/zones/Xarcabard_[S]/mobs/Zirnitra.lua

```diff
@@ -2,9 +2,40 @@
 -- Area: Xarcabard [S]
 --   NM: Zirnitra
 -----------------------------------
+local ID = zones[xi.zone.XARCABARD_S]
+mixins = { require('scripts/mixins/families/amphiptere') }
+-----------------------------------
 local entity = {}
 
+entity.onMobSpawn = function(mob)
+    -- Precise regen/regain values need retail verification.
+    -- Mob uses abilities with high frequency
+    mob:setMod(xi.mod.REGEN, 50)
+    mob:setMod(xi.mod.REGAIN, 100)
+    mob:setMod(xi.mod.WIND_ABSORB, 100)
+    mob:setMobMod(xi.mobMod.DRAW_IN, 1)
+end
+
+entity.onMobEngaged = function(mob)
+    mob:setLocalVar('hasEngaged', 1)
+end
+
+entity.onMobRoam = function(mob)
+    -- Based on video capture, mob prints this message if the party wipes.
+    -- TODO: Message below should not originate from Zirnitra herself
+    if mob:getLocalVar('hasEngaged') == 1 then
+        mob:setLocalVar('hasEngaged', 0)
+        mob:showText(mob, ID.text.THE_MONSTER_FADES)
+        DespawnMob(mob:getID())
```

Comment:
keep in mind I don't know this NM well and haven't looked it up yet, so this is generalized: Unless it despawns _immediately_ you probably  don't want to force despawn by script. We've a mobmod for handling this normally. You'd then have to move the message which I assume is seen by anyone nearby not just the party fighting it?

Reply:
Yeah the message would probably be a better fit under something like printToArea, but wouldn't use the exact message ID, which I felt more important to preserve. I looked through the available text functions for a closer fit, and seems we are a bit restricted with what we can do with the current tooling, happy to be proven wrong though :)
https://github.com/LandSandBoat/server/blob/base/src/map/lua/lua_baseentity.h#L53

Reply:
we wouldn't want printToArea in upstream, but one could custom it that way at their own server.

`showText()` does do an area message as long as the base entity (part before the `:` ) is not a player.  If its seen by bystanders, that's what I mean by area message. If its only seen by the party, might be a diff message packet than what showText does and I could possibly help research that.

If the base entity is a player and that 1st parameter is an NPC it also forces the entity to face the player.

And it uses same IDs as `messageSpecial()`

There are a few other message types that may or may not share the same ID list for the messages, Some of which we even have implementations for. I have no idea which one is retail accurate but it'd be in the packets coming to client so we could find it that way.